### PR TITLE
Fix: Do not use deprecated `set-output` command

### PR DESCRIPTION
**Pull request by localheinz**
### What is the reason for this PR?

- [x] stops using the [deprecated `set-output` command](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/)

### Author's checklist

- [x] Follow the [Contribution Guide](https://github.com/FakerPHP/Faker/blob/main/.github/CONTRIBUTING.md)
- [x] New features and changes are [documented](https://github.com/FakerPHP/fakerphp.github.io)

### Review checklist

- [ ] All checks have passed
- [ ] Changes are approved by maintainer

### CODE REVIEW

File: .github/workflows/code-coverage.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
 
       - name: Get composer cache directory
         id: composer-cache
-        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+        run: echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
```
Comment (localheinz) [inline .github/workflows/code-coverage.yml:34]: Interesting to note that there are a few variations of how dependencies are cached

- sometimes the name of the output is `dir`, sometimes `directory`
- sometimes the value is `$(composer config cache-files-dir)`, sometimes `$(composer config cache-dir)`

See #545.

**Comment by localheinz** [comment]
Tests on Windows fail - I do not know how to set the cache path on Windows.

**Comment by pimjansen** [comment]
The question i have if we even want to keep supporting windows since MS also dropped official support since 8.0

**Comment by localheinz** [comment]
@pimjansen 

I have no interest in supporting Windows.

**Comment by localheinz** [comment]
@pimjansen 

See https://github.com/FakerPHP/Faker/pull/547.

**Review by bram-pkg — APPROVED** [review]

**Comment by localheinz** [comment]
Thank you, @bram-pkg!

**Pull request merged**